### ISSUE TITLE
Question: Running Specs

### ISSUE THREAD

**Issue by r-richmond** [DavidLGoldberg/qolor#25]
So I've never done any dev on atom before but I thought I would try adding some tests that would be useful moving forward. However, I can't seem to figure out the proper setup to get rolling.

Here are my steps

```
git clone https://github.com/DavidLGoldberg/qolor.git
atom --wait qolor
[In atom | Window : Run Package Specs (Ctrl-Alt-P)]
```

After doing this every spec is failing with the error `TypeError: Cannot read property 'getBufferRange' of undefined'`

Am I missing something obvious?

![image](https://cloud.githubusercontent.com/assets/9246654/20373173/5626c4aa-ac25-11e6-82e3-97f84ef60dd1.png)

**Comment by DavidLGoldberg** [comment]
Yes.  BTW, I just read atom --wait might have some issues in windows (just making you aware).

I cd into qolor and do:
`atom -d .`

I'm sure:
`atom -d qolor` would be fine too.
You'll notice a little red box in the lower left of your editor which indicates "developer mode"

You can have a normal non dev atom open as well as an atom -d (developer) atom open simultaneously.

Another thing to be aware of is:
`apm test`

One other note.  If you don't do atom -d .  I think atom still uses your current spec files (if you have Atom open to a package)

But yes I hit control+alt+command+p to run them on the mac, but often `apm test` is more convenient when you're not debugging.  I have a few more tests done outstanding in branches sitting around, waiting for their implementation TDD style.

But chances are your tests might be for something different, so I really appreciate the effort / interest.

FYI, next on my pipeline are fixing "as" aliases for tables.  Been a little busy, but I should get some time to look at this soon.

I might warn you that the way I do testing got slightly more complicated as the index of the highlight has changed with my last attempt of enhancing for "as".

I will probably rethink this a bit.  You can probably look at the others for an example of the order they're attacked.  I don't have anything written for this, and when I looked at this the other day, I actually forgot (it's newly changed, so I'll write it up or improve it to make it more straight forward)

**Comment by DavidLGoldberg** [comment]
FYI, I don't know how familiar you are with browsers / web world but you can inspect that debugger (probably with a right click, not sure in windows, looks like Windows right?) and put breakpoints or you can enter 
`debugger` in the code and if you're in inspector mode a rerun of the scripts will break on the debugger.
